### PR TITLE
refactor(lsp): use fallback resolution in op_resolve()

### LINKED ISSUE #23339: "message": "JSX element implicitly has type 'any' because no interface 'JSX.IntrinsicElements' exists.", 	"source": "deno-ts", 	"startLineNumber": 16, 	"startColumn": 13, 	"endLineNumber": 16, 	"endColumn": 28 }]
Version: Deno 1.42.3

Captura
image

### PR DESCRIPTION
**Pull request by nayeemrmn**
Instead of using stub modules for null resolutions (#23313), make sure `op_resolve()` doesn't give null resolutions for non-erroneous cases.
- Fallback to using `CliGraphResolver::resolve()` for non-imported dependencies. This affects `jsxImportSource` which tsc will try to resolve from every module.
- Don't return a null resolution for uncached dependencies. These should load a null source instead.
- Don't return a null resolution for `node:` specifiers. Not sure why we were doing this but as far as I can see everything still works.

Fixes #23339.

### CODE REVIEW

File: cli/lsp/documents.rs [hunk 1]
```diff
@@ -1220,18 +1216,29 @@ impl Documents {
...
       } else if let Ok(npm_req_ref) =
-        NpmPackageReqReference::from_str(&specifier)
+        NpmPackageReqReference::from_str(specifier)
       {
         results.push(node_resolve_npm_req_ref(
           &npm_req_ref,
           maybe_npm,
           referrer,
         ));
+      } else if let Ok(specifier2) = self.get_resolver().resolve(
```
Comment (dsherret) [inline cli/lsp/documents.rs:1231]: Maybe `resolved_specifier`?

**Review by dsherret — COMMENTED** [review]
LGTM, but let's land this after this release so we have some time to test it.

File: cli/lsp/documents.rs (hunk 1 above)
Comment (nayeemrmn) [reply cli/lsp/documents.rs:1231]: Oops, that was from debugging. I'll change it back to `specifier`.

**Review by dsherret — APPROVED** [review]
LGTM

**Pull request merged**